dts/arm/realtek/ec/rts5912.dtsi

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2024 Realtek Semiconductor Corporation, SIBG-SD7
+ *
+ */
+
+#include <arm/armv8-m.dtsi>
+#include <zephyr/dt-bindings/clock/rts5912_clock.h>
+#include <zephyr/dt-bindings/gpio/realtek-gpio.h>
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			compatible = "arm,cortex-m33f";
+			reg = <0>;
+			cpu-power-states = <&idle &suspend_to_ram>;
+		};
+
+		power-states {
+			idle: idle {
+				compatible = "zephyr,power-state";
+				power-state-name = "suspend-to-idle";
+				min-residency-us = <100000>;
+			};
+
+			suspend_to_ram: suspend_to_ram {
+				compatible = "zephyr,power-state";
+				power-state-name = "suspend-to-ram";
+				min-residency-us = <250000>;
+			};
+		};
+	};
+
+	flash0: flash@20000400 {
+		reg = <0x20000400 0x4FC00>;
+	};
+
+	sram0: memory@20050000 {
+		compatible = "mmio-sram";
+		reg = <0x20050000 0x8000>;
+	};
+
+	soc {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "simple-bus";
+		ranges;
+	};
+
+	swj_port: swj_port {
+		compatible = "swj-connector";
+		pinctrl-0 = < &jtag_tdi_gpio87
+				&jtag_tdo_gpio88
+				&jtag_rst_gpio89
+				&jtag_clk_gpio90
+				&jtag_tms_gpio91 >;
+		pinctrl-names = "default";
+	};
```

Comment:
Move the `swj-port` node the file `debug_swj.c`  into a separate commit so that if follows the pinctrl support.

Reply:
We move those codes which related to debug_swj into the new commit after pinctrl named 
"soc: realrek: ec: Add debug_swj initial version of RTS5912.".